JavaBytecodeCompiler/src/main/java/org/openzen/zenscript/javabytecode/compiler/JavaForeachWriter.java

```diff
@@ -90,6 +90,8 @@ private void handleArray(final int z, final JavaLocalVariableInfo arrayTypeInfo)
 		javaWriter.dup();
 		javaWriter.loadInt(z);
 
+		// ToDo: atm, statement.list.type can also be Optional[ArrayTypeId]
+		//   Does this need to be handled here or should the call already be wrapped in an unwrapOptional expression?
```

Comment:
What the ToDo says: Do we need an if optional check, or should this already be done at an earlier stage of the compiling process?

Reply:
I'd think that if it's an optional, we have to check for it. We'd just need to decide if it should skip the iteration, or throw an exception.